unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs

```diff
@@ -18,6 +18,8 @@ public class ECSScenesUiSystem : IDisposable
         private readonly BaseList<IParcelScene> loadedScenes;
         private readonly BaseVariable<bool> loadingHudVisibleVariable;
 
+        private static HashSet<IParcelScene> scenesToSort = new HashSet<IParcelScene>();
```

Comment:
If we want to remove the allocation of the hashset, I wouldn't add that field as static.
I think it would probably be better to set it as part of the class state and modify `ApplyParenting` to receive it as argument

Reply:
Agree! No idea why I didn't do it this way originally, good spot